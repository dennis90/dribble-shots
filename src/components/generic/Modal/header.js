import React from 'react'
import {Link} from 'react-router-dom'

const Header = ({closeRoute}) => {
  return (
    <div className="modal-header">
      <Link to={closeRoute}>
        <span className="close-modal-button"></span>
      </Link>
    </div>
  )
}

export default Header
