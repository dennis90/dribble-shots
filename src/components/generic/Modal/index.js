import React from 'react'
import PropTypes from 'prop-types'

import Header from './header'
import '../../../assets/css/modal.css'

const Modal = ({children, closeRoute}) => {
  return (
    <div className="modal-container">
      <div className="modal">
        <Header closeRoute={closeRoute} />
        {children}
      </div>
      <div className="modal-background" />
    </div>
  )
}

Modal.propTypes = {
  closeRoute: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired
}

export default Modal
