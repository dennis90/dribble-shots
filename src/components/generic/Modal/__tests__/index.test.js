import React from 'react'
import {shallow} from 'enzyme'

import Modal from '../index'

describe('Modal Component', () => {
  it('should render modal structure', () => {
    const wrapper = shallow(
      <Modal closeRoute={'/'}>
        <span>Modal Children</span>
      </Modal>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
