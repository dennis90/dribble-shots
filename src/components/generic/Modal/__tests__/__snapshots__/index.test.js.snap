// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render modal structure 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    closeRoute="/"
>
    <span>
        Modal Children
    </span>
</Modal>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="modal"
>
          <Header
                    closeRoute="/"
          />
          <span>
                    Modal Children
          </span>
</div>,
        <div
          className="modal-background"
/>,
      ],
      "className": "modal-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              closeRoute="/"
/>,
            <span>
              Modal Children
</span>,
          ],
          "className": "modal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "closeRoute": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Modal Children",
            },
            "ref": null,
            "rendered": "Modal Children",
            "type": "span",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "modal-background",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="modal"
>
            <Header
                        closeRoute="/"
            />
            <span>
                        Modal Children
            </span>
</div>,
          <div
            className="modal-background"
/>,
        ],
        "className": "modal-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Header
                closeRoute="/"
/>,
              <span>
                Modal Children
</span>,
            ],
            "className": "modal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "closeRoute": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Modal Children",
              },
              "ref": null,
              "rendered": "Modal Children",
              "type": "span",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "modal-background",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
